jaypeasee/whats-cookin
# Refactor - Classes

# Pull Request Template:


#### What's this PR do?
1. Removes the `Ingredient` class and test for it because they are unnecessary.
1. Moves a method that was in `scripts.js` to the `Pantry` class and tested it.
1. Refactors a few methods in the `User` Class
1. Refactors the variable declarations in the `User` Class and `Recipe` Class to not instantiate `Ingredient` anymore.


#### Where should the reviewer start?

1. Review `pantry-test.js` and `pantry.js`
1. Review `recipe-test.js`
1. Review `user-test.js` and `user-js`
1. Review `scripts.js`


#### How should this be manually tested?
1. Run `npm test` to confirm all tests are passing
1. Decide to cook a recipe and look at the `currentUser.recipesToCook` array
1. View that the pantry is still loading the user's ingredients.

#### Any background context you want to provide?
* This was a pretty major refactor on the classes, but it is likely there may need to be a little more in the `Recipe` Class down the line.
* We now need to display the `recipesToCook` array on the `DOM`.
